pkg/plugins/helm/provider.go

```diff
@@ -70,21 +106,367 @@ func (p *Provider) getHelmClient() (helmClient *helm.Client, err error) {
+		tdj := getTaskDirectiveJson(td)
+
+		task := models.NewTask("", job.JobId, "", "kubernetes", constants.ActionCreateCluster, tdj, job.Owner)
+		tasks = append(tasks, task)
+		tl := models.TaskLayer{
+			Tasks: tasks,
+			Child: nil,
+		}
+
+		return &tl, nil
+	case constants.ActionUpgradeCluster:
+		tasks := make([]*models.Task, 1)
+
+		td := TaskDirective{VersionId: job.VersionId, ClusterId: job.ClusterId, RuntimeId: jodDirective.RuntimeId}
+		tdj := getTaskDirectiveJson(td)
+
+		task := models.NewTask("", job.JobId, "", "kubernetes", constants.ActionUpgradeCluster, tdj, job.Owner)
+		tasks = append(tasks, task)
+		tl := models.TaskLayer{
+			Tasks: tasks,
+			Child: nil,
+		}
+
+		return &tl, nil
+	case constants.ActionRollbackCluster:
+		tasks := make([]*models.Task, 1)
+
+		td := TaskDirective{ClusterId: job.ClusterId, RuntimeId: jodDirective.RuntimeId}
+		tdj := getTaskDirectiveJson(td)
+
+		task := models.NewTask("", job.JobId, "", "kubernetes", constants.ActionRollbackCluster, tdj, job.Owner)
+		tasks = append(tasks, task)
+		tl := models.TaskLayer{
+			Tasks: tasks,
+			Child: nil,
+		}
+
+		return &tl, nil
+	case constants.ActionDeleteClusters:
+		tasks := make([]*models.Task, 1)
+
+		td := TaskDirective{ClusterId: job.ClusterId, RuntimeId: jodDirective.RuntimeId}
+		tdj := getTaskDirectiveJson(td)
+
+		task := models.NewTask("", job.JobId, "", "kubernetes", constants.ActionDeleteClusters, tdj, job.Owner)
+		tasks = append(tasks, task)
+		tl := models.TaskLayer{
+			Tasks: tasks,
+			Child: nil,
+		}
+
+		return &tl, nil
+	case constants.ActionCeaseClusters:
+		tasks := make([]*models.Task, 1)
+
+		td := TaskDirective{ClusterId: job.ClusterId, RuntimeId: jodDirective.RuntimeId}
+		tdj := getTaskDirectiveJson(td)
+
+		task := models.NewTask("", job.JobId, "", "kubernetes", constants.ActionCeaseClusters, tdj, job.Owner)
+		tasks = append(tasks, task)
+		tl := models.TaskLayer{
+			Tasks: tasks,
+			Child: nil,
+		}
+
+		return &tl, nil
+	default:
+		return nil, fmt.Errorf("the job action [%s] is not supported", job.JobAction)
+	}
 	return nil, nil
 }
+
 func (p *Provider) HandleSubtask(task *models.Task) error {
+	taskDirective, err := getTaskDirective(task.Directive)
+	if err != nil {
+		return err
+	}
+
+	hc, err := p.getHelmClient(taskDirective.RuntimeId)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	switch task.TaskAction {
+	case constants.ActionCreateCluster:
+		appc, err := appclient.NewAppManagerClient(ctx)
+		if err != nil {
+			return err
+		}
+
+		req := pb.GetAppVersionPackageRequest{
+			VersionId: utils.ToProtoString(taskDirective.VersionId),
+		}
+
+		resp, err := appc.GetAppVersionPackage(ctx, &req)
+		if err != nil {
+			return err
+		}
+
+		pkg := resp.GetPackage()
+		r := bytes.NewReader(pkg)
+
+		c, err := chartutil.LoadArchive(r)
+		if err != nil {
+			return err
+		}
+
+		_, err = hc.InstallReleaseFromChart(c, taskDirective.Namespace,
+			helm.ValueOverrides([]byte(c.Values.GetRaw())),
```

Comment:
If values is changed by user when create cluster, here need to set new values into c.Values

Reply:
I think I need a field on Cluster to store the values.